swap_meet/item.py

```diff
@@ -1,38 +1,28 @@
-
+from .constants import CONDITION_STATES
 import uuid
+import math
 
 class Item:
     def __init__(self, id=None, condition=0, age=None):
-        if id is not None and isinstance(id, int):
-            self.id = id
-        else:
-            self.id = uuid.uuid4().int  # long unique number
+        if id is not None and not isinstance(id, int):
+            raise TypeError("ID has to be an integer or not provided, so it can be safely set!")
+        self.id = id if id is not None else uuid.uuid4().int # long unique number  
 
-        if isinstance(condition, (int, float)) and 1.0 <= condition <= 5.0:
-            self.condition = condition
-        else:
-            self.condition = 0
+        if not isinstance(condition, (int, float)):
+            raise TypeError("Condition has to be a number between 0.0 and 5.0")
+        if not (0.0 <= condition <= 5.0):
+            raise ValueError("Condition has to be a number between 0.0 and 5.0")
+        self.condition = condition
             
         self.age = age
 
     def get_category(self):
-        return self.__class__.__name__
+        return "Item"
     
     def __str__(self):
         return f"An object of type {self.get_category()} with id {self.id}."
     
     def condition_description(self):
-        if self.condition == 0:
-            return "Unusable"
-        elif self.condition == 1:
-            return "Poor"
-        elif self.condition == 2:
-            return "Fair"
-        elif self.condition == 3:
-            return "Good"
-        elif self.condition == 4:
-            return "Very good"
-        elif self.condition == 5:
-            return "Like new"
+        return CONDITION_STATES[math.floor(self.condition + 0.5)]
```

Comment:
👍  This implements a typical "round" behavior, which will handle all those pesky float conditions! The built-in python `round()` can seem a bit different compared to what we learn in school, though keep in mind that because of floating point error, it's even possible for this approach to produce unexpected results.

Note that for positive numbers, `math.float()` and `int()` produce the same result. So here, we might more expect to see
```py
        return CONDITION_STATES[int(self.condition + 0.5)]
```
which avoids the need to import `math`. Technically, there were still no library imports allowed in this project, but  since this is really just equivalent to `int()` here, it's fine.